scl/cri/cri.conf

```diff
@@ -0,0 +1,51 @@
+#########################################################
+# Parser for CRI log format
+# ISO_TIME(nano) SP STREAM SP TAG SP MSG
+# https://github.com/kubernetes/kubernetes/issues/44976
+#========================================================
+# Retrieves information about K8S meta from log filename:
+#  - container ID
+#  - container Name
+#  - POD NameSpace
+#  - POD Name
+# Use .cri prefix
+# Set .cri.time as S_ISODATE with use of date-parser()
+#
+# Note: does not support multiline, line = message
+#########################################################
+# Example log:
+# 2022-02-04T18:14:43.219493781+01:00 stdout F Starting up on port 80
+
+@define crio-log-regexp "([a-z0-9-]+)_([a-z0-9-]+)_([a-z0-9]+)-([a-z0-9]{64}).log"
+
+block rewrite extract-container-id (value("MESSAGE")){
+    # make the container-id of 12 characters long as usual in cli
+    subst("`crio-log-regexp`", "$(substr $4 0 12)", value("`value`"), flags(global, store-matches), type(pcre));
+};
+
+block parser cri-parser(prefix('.cri.')) {
+    channel {
+        parser {
+            csv-parser(
+                columns("`prefix`time", "`prefix`stream", "`prefix`flag", "`prefix`message")
```

Comment:
These are all part of the $MESSAGE macro? What is in the syslog header in this case?

Could you show an example?

Reply:
It's file source with no-parse flag set.
https://github.com/peterviskup/syslog-ng-container/blob/main/syslog-ng/conf.d/10-source-containers.conf#L7